### PR TITLE
Updating template with all the datum learnings

### PR DESCRIPTION
**Pull request by golanglemonade**
- Includes an example `todo` schema

server starts up with basic setup:

```
 task run-dev
task: [docker:redis] docker compose -f ./docker/docker-compose-redis.yml -p redis up -d
[+] Building 0.0s (0/0)                                                                                                                                         docker:desktop-linux
[+] Running 1/0
 ✔ Container redis  Running                                                                                                                                                     0.0s 
task: [docker:fga] docker compose -f ./docker/docker-compose-fga.yml -p fga up -d
[+] Building 0.0s (0/0)                                                                                                                                         docker:desktop-linux
[+] Running 3/3
 ✔ Container postgres  Healthy                                                                                                                                                  0.0s 
 ✔ Container migrate   Exited                                                                                                                                                   0.0s 
 ✔ Container openfga   Running                                                                                                                                                  0.0s 
task: [run-dev] go run main.go serve --debug --pretty
2024-03-01T17:46:01.757-0700	INFO	marionette/marionette.go:95	task manager running
2024-03-01T17:46:01.757-0700	DEBUG	otelx/otel.go:67	Tracing disabled
2024-03-01T17:46:01.757-0700	INFO	marionette/scheduler.go:101	scheduler running
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	driver.Query: query=PRAGMA foreign_keys args=[]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	driver.Query: query=PRAGMA foreign_keys args=[]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	driver.Exec: query=PRAGMA foreign_keys = off args=[]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	driver.Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa): started
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Query: query=PRAGMA foreign_key_check args=[]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Query: query=SELECT sqlite_version() args=[]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Query: query=SELECT name FROM pragma_collation_list() args=[]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Query: query=SELECT `name`, `file` FROM pragma_database_list() WHERE `name` IN (?) args=[main]
2024-03-01T17:46:01.758-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Query: query=
SELECT
	sqlite_master.name, sqlite_master.sql, wr, strict
FROM
	sqlite_master
	JOIN pragma_table_list(sqlite_master.name)
WHERE
	sqlite_master.type = 'table'
	AND sqlite_master.name NOT LIKE 'sqlite_%'
	AND sqlite_master.name NOT LIKE 'libsql_%'
 AND sqlite_master.name IN (?) args=[todos]
2024-03-01T17:46:01.759-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Exec: query=CREATE TABLE `todos` (`id` text NOT NULL, `name` text NOT NULL, `description` text NULL, PRIMARY KEY (`id`)) args=[]
2024-03-01T17:46:01.759-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Exec: query=CREATE UNIQUE INDEX `todo_name` ON `todos` (`name`) args=[]
2024-03-01T17:46:01.759-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa).Query: query=PRAGMA foreign_key_check args=[]
2024-03-01T17:46:01.759-0700	DEBUG	ent	dialect/dialect.go:79	Tx(60e27236-f138-41ea-bf3a-e9978e76e5fa): committed
2024-03-01T17:46:01.759-0700	DEBUG	ent	dialect/dialect.go:79	driver.Exec: query=PRAGMA foreign_keys = on args=[]
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/ready", "method": "GET"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/query", "method": "POST"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/query", "method": "GET"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/playground", "method": "GET"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/playground/playground.css", "method": "GET"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/playground/playground.js", "method": "GET"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/playground/favicon.png", "method": "GET"}
2024-03-01T17:46:01.761-0700	INFO	server/server.go:78	registered route	{"route": "/playground/logo.png", "method": "GET"}
2024-03-01T17:46:02.168-0700	INFO	echozap@v0.0.0-20231205193458-b29cc54cd34c/logger.go:51	Success	{"remote_ip": "127.0.0.1", "latency": "6.77675ms", "host": "localhost:1337", "request": "GET /ready", "status": 200, "size": 64, "user_agent": "curl/8.1.2", "request_id": "AZkbrQWHywkCMroVPfodjtAlDoLJWuiR"}
{"time":"2024-03-01T17:46:02.168518-07:00","level":"INFO","prefix":"echo","message":"remote_ip=127.0.0.1, method=GET, uri=/ready, status=200, session=, host=localhost:1337, referer=, user_agent=curl/8.1.2, route=/ready, path=/ready, auth=\n"}
```

version cli command: 

```
✗ go run cmd/cli/main.go version
Version: no-info
Go Version: go1.22.0
Git Commit: 
Commit Date: 
OS/Arch: darwin/arm64
Dirty: false
User Agent: datum-cli/no-info (darwin) arm64 (go1.22.0)
```

ready + livez handlers included
```
 curl localhost:1337/ready
{
  "status": {
    "db_primary": "OK",
    "redis": "OK"
  }
}
```

```
curl localhost:1337/livez
{
  "status": "UP"
}
```

### CODE REVIEW

File: .github/workflows/releaser.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
+name: Release
```
Comment (matoszz) [inline .github/workflows/releaser.yml:1]: given the frequency at which we're re-using this pipeline, we should probably make a dedicated repo for it and call it via the dispatch action

File: internal/ent/entc.go [hunk 2]
```diff
@@ -6,23 +6,72 @@ package main
...
+	"github.com/datumforge/fgax/entfga"
 	"github.com/ogen-go/ogen"
+	"go.uber.org/zap"
+	"gocloud.dev/secrets"
+
+	"github.com/datumforge/entx"
+
+	"github.com/datumforge/datum/pkg/analytics"
+	"github.com/datumforge/datum/pkg/utils/emails"
+	"github.com/datumforge/datum/pkg/utils/marionette"
```
Comment (matoszz) [inline internal/ent/entc.go:26]: we probably don't want to import these packages from datum into the template give we'd likely move them any way?

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,16 @@
+sonar.projectKey=datumforge_datum
```
Comment (matoszz) [inline sonar-project.properties:1]: might want to at least change this to use the template name vs. datum?

Comment (golanglemonade) [reply sonar-project.properties:1]: `datumforge_datum` is the overall "project" name - e.g. more the "org level" - `sonar.projectName` is the actual project name, which is already updated

Comment (golanglemonade) [reply sonar-project.properties:1]: jk, I think I know whats wrong here

**Review by matoszz — DISMISSED** [review]

**Review by matoszz — APPROVED** [review]

**Pull request merged**